veros/core/external/solvers/scipy.py

```diff
@@ -90,115 +101,45 @@ def _jacobi_preconditioner(state, matrix):
 
     @staticmethod
     def _assemble_poisson_matrix(state):
-        """
-        Construct a sparse matrix based on the stencil for the 2D Poisson equation.
-        """
         vs = state.variables
         settings = state.settings
 
-        boundary_mask = ~npx.any(vs.boundary_mask, axis=2)
+        diags, offsets = assemble_poisson_matrix(state)
+        offsets = tuple(-dx * diags[0].shape[1] - dy for dx, dy in offsets)
```

Comment:
Could you add some comments to this? Like why we flatten the offsets, and why we add the wrap diags.